### PR TITLE
Fix lexer issue with 'e' following integers across lines

### PR DESCRIPTION
**Pull request by esleche**
Fixes https://github.com/hylo-lang/hyloc/issues/873

### CODE REVIEW

File: Sources/FrontEnd/Parse/Lexer.swift [hunk 1]
```diff
@@ -27,6 +27,13 @@ public struct Lexer: IteratorProtocol, Sequence {
     while true {
       if index == sourceCode.text.endIndex { return nil }
 
+      // Skip newlines.
+      if sourceCode.text[index].isNewline {
+        previousTokenWasInt = false
```
Comment (kyouko-taiga) [inline Sources/FrontEnd/Parse/Lexer.swift:32]: It's a little suspicious that you can unset `preciousTokenWasInt` here but not after any other whitespace. Why is that?

**Review by kyouko-taiga — CHANGES_REQUESTED** [review]
It's a bit overkill to create end-to-end tests only to check that the lexer is doing its work properly. It would be better to write these tests as parser tests (see `HyloTests/TestCases/Parsing`) or, if type checking is needed, as type checking tests (see `HyloTests/TestCases/TypeChecking`).

File: Sources/FrontEnd/Parse/Lexer.swift (hunk 1 above)
Comment (esleche) [reply Sources/FrontEnd/Parse/Lexer.swift]: Done 👍

**Review by kyouko-taiga — APPROVED** [review]
Nice changes, thanks!
LGTM.

**Pull request merged**